pkg/handler/handler.go

```diff
@@ -433,9 +433,10 @@ func (m *Modifier) buildPodPatchConfig(pod *corev1.Pod) *podPatchConfig {
 	}
 
 	// Use the STS WebIdentity method if set
-	request := cache.Request{Namespace: pod.Namespace, Name: pod.Spec.ServiceAccountName, RequestNotification: true}
+	gracePeriodEnabled := m.saLookupGraceTime > 0
+	request := cache.Request{Namespace: pod.Namespace, Name: pod.Spec.ServiceAccountName, RequestNotification: gracePeriodEnabled}
```

Comment:
I added this change to toggle `RequestNotification` only when the grace period is enabled.

Previously it was basically a no-op when the feature is disabled and `RequestNotification` is true, but now we're fetching from the API server when it's true. So we only want it set when the feature is enabled.